.azure-pipelines/buildAndPackage.yml

```diff
@@ -30,10 +30,37 @@ pr: none
 pool:
   vmImage: windows-latest
 
-steps:
-- template: templates/checkout-and-credscan.yml
-- template: templates/install-java.yml
-- template: templates/secure-files.yml
-- template: templates/build-and-coverage.yml
-- template: templates/publish-artefacts.yml
-- template: templates/alert-failure.yml
+stages:
+- stage: Build-Stage
+  jobs:
+  - job: Build
+    steps:
+    - template: templates/build-stage/checkout-and-credscan.yml
+    - template: templates/build-stage/install-java.yml
+    - template: templates/build-stage/secure-files.yml
+    - template: templates/build-stage/build-and-coverage.yml
+    - template: templates/build-stage/publish-artefacts.yml
+    - template: templates/alert-failure.yml
+- stage: Maven-Preview-Stage
+  jobs: 
```

Comment:
this stage probably needs a condition (if) to only execute when the pipeline is triggered from the dev branch